lib/services/appmanager-service.ts

```diff
@@ -99,5 +139,31 @@ class AppManagerService implements IAppManagerService {
 			}
 		}).future<void>()();
 	}
+
+	private findGroups(identityStrings:string[]): IFuture<string[]> {
+		return ((): any => {
+			let availableGroups = this.$server.tam.getGroups().wait();
+			let groupIds: string[] = [];
+
+			if (!availableGroups.length) {
+				this.$errors.failWithoutHelp("No available groups found.");
+			}
+
+			_.forEach(identityStrings, (identityStr, index) => {
```

Comment:
I think you can use `_.map` here ( https://lodash.com/docs#map ) 
```TypeScript
return _.map(identityStrings, identityStr => {
	let group = helpers.findByNameOrIndex(identityStr, availableGroups, (group) => group.Name);

	if (!group) {
		this.$errors.fail("Could not find group named '%s' or was not given a valid index.%sTo see available groups run $ appbuilder appmanager groups",
			identityStr,
			os.EOL);
	}
	
	return group.Id;
});
```

This way you can omit groupIds declaration at the beginning of the method.